app/services/aptrust/aptrust_service.rb

```diff
@@ -0,0 +1,350 @@
+                                             "status_history=#{status_history}",
+                                             "" ], bold_puts: false if aptrust_service_debug_verbose
+
+      status_history = track_deposit( id: work.id, status: 'bagging', status_history: status_history )
+      working_dir ||= dir_working
+      id ||= identifier( work: work )
+      working_dir = File.join( working_dir, id )
+      ::Deepblue::LoggingHelper.bold_debug [ ::Deepblue::LoggingHelper.here,
+                                             ::Deepblue::LoggingHelper.called_from,
+                                             "working_dir=#{working_dir}",
+                                             "work=#{work}",
+                                             "id=#{id}",
+                                             "" ], bold_puts: false if aptrust_service_debug_verbose
+      Dir.mkdir( working_dir ) unless Dir.exist? working_dir
+      bag = BagIt::Bag.new( working_dir )
+      bag_info = bag_info( work: work )
+      bag.write_bag_info( bag_info ) # Create bagit-info.txt file
+      aptrust_info_write( bag: bag, work: work )
+      bag_data_dir = File.join( bag.bag_dir, "data" )
+      bag_data_dir = Pathname.new bag_data_dir
+      export_work( work: work, target_dir: bag_data_dir )
+      bag_manifest( bag: bag )
+      status_history = track_deposit( id: work.id, status: 'bagged', status_history: status_history )
+      working_dir
+    end
+
+    def self.bag_date_now()
+      rv = Time.now.utc.strftime("%Y-%m-%dT%H:%M:%SZ")
+      rv = Time.parse(rv).iso8601
+      return rv
+    end
+
+    def self.bag_info( work: )
+      # add bagit-info.txt file
+      # The length of the following 'internal_sender_description' does not work with the current bagit gem, maybe later.
+      # pub = monograph_presenter.publisher.blank? ? '' : monograph_presenter.publisher.first.squish[0..55]
+      # internal_sender_description = "This bag contains all of the data and metadata in a Monograph from #{pub} which has been exported from the Fulcrum publishing platform hosted at www.fulcrum.org."
+      {
+        BAG_INFO_KEY_SOURCE => BAG_INFO_VALUE_SOURCE,
+        BAG_INFO_KEY_COUNT => '1',
+        BAG_INFO_KEY_DATE => bag_date_now(),
+        'Internal-Sender-Description' => bag_description( work ),
+        'Internal-Sender-Identifier' => id_work( work )
+      }
+    end
+
+    def self.bag_description( work )
+      "Bag for a #{work.class.name} hosted at deepblue.lib.umich.edu/data/" # TODO: improve this, or move to config
+    end
+
+    def self.bag_manifest( bag: )
+      bag.manifest!(algo: 'md5') # Create manifests
+
+      # HELIO-4380 demo.aptrust.org doesn't like this file for some reason, gives an ingest error:
+      # "Bag contains illegal tag manifest 'sha1'""
+      # APTrust only wants SHA256, or MD5, not SHA1.
+      # 'tagmanifest-sha1.txt' is a bagit gem default, so we need to remove it manually.
```

Comment:
So you can't specify SHA256 or MD5 for the tagmanifest?

Reply:
Apparently not.